src/roman_datamodels/datamodels.py

```diff
@@ -322,6 +326,174 @@ class GuidewindowModel(DataModel):
     pass
 
 
+class ModelContainer(DataModel, Sequence):
+    # Needs a number of methods to properly handle the contents
+    def __init__(self, init=None, asn_schema= None, asn_file_path=None, model_file_path=None, iscopy=False, **kwargs):
+        # __init__(self, init=None, asn_exptypes=None, asn_n_members=None,
+        #                  iscopy=False, **kwargs):
+        super().__init__(init=None, **kwargs)
+
+        self._models = []
+        self._iscopy = iscopy
+        # self.asn_exptypes = asn_exptypes
+        # self.asn_n_members = asn_n_members
+        # self.asn_table = {}
+        # self.asn_table_name = None
+        # self.asn_pool_name = None
+
+        self._memmap = kwargs.get("memmap", False)
+        self._return_open = kwargs.get('return_open', True)
+        self._save_open = kwargs.get('save_open', True)
+
+        if asn_file_path:
+            with builtins.open(asn_file_path, 'r') as asn_file:
+                asn_schema = json.load(asn_file)
+
+            for product in asn_schema['products']:
+                for member in product['members']:
```

Comment:
So the model will try and open *all* members in an association? This might be a memory issue right off, some of my test associations have 32 members and reading all of these into memory at once may be problematic.